src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java

```diff
@@ -141,16 +145,59 @@ private CallbackResponse populateRespondentCopyObjects(CallbackParams callbackPa
     }
 
     private CallbackResponse validateRespondentWitnesses(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        if (featureToggleService.isMultipartyEnabled()) {
+            if (solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORONE)) {
+                return validateWitnesses(callbackParams.getCaseData().getRespondent1DQ());
+            } else if (solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORTWO)) {
+                return validateWitnesses(callbackParams.getCaseData().getRespondent2DQ());
+            } else if (respondent2HasSameLegalRep(caseData)) {
+                if (caseData.getRespondentResponseIsSame() != null && caseData.getRespondentResponseIsSame() == NO) {
```

Comment:
We are checking this several times:
`caseData.getRespondentResponseIsSame() != null && caseData.getRespondentResponseIsSame() == NO`

Recommend extracting to a common method for readability + code clenliness.
Used by mothods:
- validateRespondentWitnesses
- validateRespondentExperts
- validateUnavailableDates
- setApplicantResponseDeadlineV1